File: backend/java-services/file-service/src/main/java/com/ragflow/file/dto/response/ApiResponse.java
package com.ragflow.file.dto.response;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Builder;

@Builder
@JsonInclude(JsonInclude.Include.NON_NULL)
public record ApiResponse<T>(
        boolean success,
        @JsonProperty("status_code") int statusCode,
        String error,
        T data
) {
    public static <T> ApiResponse<T> success(int statusCode, T data) {
        return new ApiResponse<>(true, statusCode, null, data);
    }

    public static <T> ApiResponse<T> error(int statusCode, String error) {
        return new ApiResponse<>(false, statusCode, error, null);
    }
}

File: backend/node-services/parser-service/src/common/types/apiResponse.ts
export type ApiSuccessResponse<DATA> = {
  success: true;
  status_code: number;
  data: DATA;
};

export type ApiErrorResponse<ERROR = string> = {
  success: false;
  status_code: number;
  error: ERROR;
};

export type ApiResponse<DATA, ERROR = string> =
  ApiSuccessResponse<DATA> | ApiErrorResponse<ERROR>;
